src/telliot_feed_examples/reporters/reporter_autopay_utils.py

```diff
@@ -48,46 +48,41 @@ async def get_single_tip(
         msg = "can't suggest single tip for autopay contract not connected"
         error_status(note=msg, log=logger.critical)
         return None
-    query_id_lis, status = await autopay.read("getFundedQueryIds")
+
+    current_time = TimeStamp.now().ts
+
+    tips, status = await autopay.read("getPastTips", _queryId=query_id)
     if not status.ok:
-        msg = "unable to read getFundedQueryIds from autopay"
+        msg = "unable to read getPastTipCount in autopay"
         error_status(note=msg, log=logger.warning)
         return None
-    current_time = TimeStamp.now().ts
-    if query_id in query_id_lis:
-        tips, status = await autopay.read("getPastTips", _queryId=query_id)
+    count = len(tips)
+    if count > 0:
```

Comment:
Could we either add a comment explaining what this binary search is looking for, or just iterate through the entire list since we're not trying to save on gas in Python (given that this algo is from the Solidity code in the Autopay contract)?